drivers/comparator/qm_comparator.c

```diff
@@ -79,6 +81,9 @@ int qm_ac_set_config(const qm_ac_config_t *const config)
 	QM_SCSS_CMP->cmp_ref_pol = config->polarity;
 	QM_SCSS_CMP->cmp_pwr = config->power;
 
+	/* Provide some delay after power Change to Comparator*/
+	clk_sys_udelay(5 * ONE_MICROSECOND);
```

Comment:
Reading it back is not helping. 
Issue still reproducible even after reading it after write.